web/src/components/Both.js

```diff
@@ -1,11 +1,13 @@
 import React, { Component } from "react";
 import { SketchPicker } from 'react-color';
+// CR * formatting
 
 
 
 export class Both extends Component {
     constructor(props) {
         super(props);
+        // CR * grouping properties
```

Comment:
There are a lot of properties on this state, which is okay. You may benefit from putting less top-level properties and more nested objects on the state in order to organize related properties.